target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -178,6 +178,12 @@ case "$FIRMWARE" in
 	cf-e385ac)
 		ath10kcal_extract "art" 20480 12064
 		;;
+	k2t)
+		ath10kcal_extract "art" 20480 12064
+		ath10kcal_patch_mac $(macaddr_add $(head -c400 $(find_mtd_chardev config) | grep -o wan_mac.* | cut -d\" -f3) +2)
+		ln -sf /lib/firmware/ath10k/pre-cal-pci-0000\:00\:00.0.bin \
+			/lib/firmware/ath10k/QCA9888/hw2.0/board.bin
+		;;
```

Comment:
If the mac address patched here isn't working, we might also need to recalculate the checksum value inside ART data.
Maybe you could try ath10kcal_patch_mac_crc function in target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
(But I'm not sure if the checksum location are the same.)

Reply:
my test shows that the patch work on K2T A3 but not on K2T A1

Reply:
BTW If the patched mac isn't working it could mean that the firmware rejected the entire calibration data due to incorrect checksum. So if it doesn't work, we should remove this mac address patch as it breaks the checksum.